addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2384,6 +2401,11 @@ const SnippetOptionWidget = Widget.extend({
                 });
                 return activeClassNames;
             }
+            case 'displayOnMobile': {
+                const classList = Array.prototype.slice.call(this.$target[0].classList);
+                return classList.includes("d-none") &&
+                classList.some(className => className.startsWith("d-md-")) ? 'hide' : 'show';
```

Comment:
It's kinda strange to have the 'hide' and 'show' value duplicated between XML and JS. XML is there to ease describing option without JS...once you need JS, it is strange to rely on XML data for this. Instead of `data-display-on-mobile` with two values, do this: `data-hide-on-mobile="true"`.

Then you can do like this line of code: `return this.$target[0].querySelector('.description') ? 'true' : '';`. And in the method, instead of cheching if 'hide' or 'show', you can just check if `widgetValue` is truthy. One day we'll improve to allow to understand boolean values in a more easy way.